packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/listFeatures.ts

```diff
@@ -68,30 +69,50 @@ const ListStyleDefinitionMetadata = createObjectDefinition<ListStyleMetadata>(
     true /** allowNull */
 );
 
+const shouldHandleIndentationEvent = (indenting: boolean) => (
+    event: PluginKeyboardEvent,
+    editor: IEditor
+) => {
+    const { keyCode, altKey, shiftKey, ctrlKey, metaKey } = event.rawEvent;
+    return (
+        !ctrlKey &&
+        !metaKey &&
+        (keyCode === Keys.TAB
+            ? !altKey && shiftKey === !indenting
+            : shiftKey && altKey && keyCode === (indenting ? Keys.RIGHT : Keys.LEFT)) &&
+        cacheGetListElement(event, editor)
+    );
+};
+
+const handleIndentationEvent = (indenting: boolean) => (
+    event: PluginKeyboardEvent,
+    editor: IEditor
+) => {
+    const isRTL =
+        event.rawEvent.keyCode !== Keys.TAB &&
+        getComputedStyle(editor.getElementAtCursor(), 'direction') == 'rtl';
```

Comment:
getComputedStyle() is kind of heavy. So we should only call it when we really need to.

In this case, it means when we know that it is LEFT/RIGHT key with SHIFT key pressed, and there is list, then finally we can check if it is RTL. So that this function is only called when really need.

Reply:
Shift key check is handled by `shouldHandleIndentationEvent`, getComputedStyle doesn't get called if keycode is TAB